This pull request is failing in two of or GH continuous integration tests:  those for Windows Mingw64 pull request and push:

* push request: https://github.com/Perl/perl5/actions/runs/6835969949/job/18590440823?pr=21629
* pull: https://github.com/Perl/perl5/actions/runs/6835967877/job/18609019538?pr=21629

I've rerun one of the two runs with no improvement.  I believe this is a Perl build-time error on these machines -- not an error in Getopt-Long-2.57.  The failing code looks like this:
```
2023-11-13T02:04:34.4411123Z ..\miniperl.exe -I..\lib ..\make_ext.pl "MAKE=gmake" --dir=..\cpan --dir=..\dist --dir=..\ext lib
2023-11-13T02:04:34.4424505Z Generating a gmake-style Makefile
2023-11-13T02:04:34.4805196Z Writing Makefile for lib
2023-11-13T02:04:34.4809737Z gmake[1]: Entering directory 'D:/a/perl5/perl5/dist/lib'
2023-11-13T02:04:34.5832401Z ..\miniperl.exe -I..\lib ..\mkppport
2023-11-13T02:04:34.6011152Z Can't locate base.pm in @INC (you may need to install the base module) (@INC entries checked: D:\a\perl5\perl5\lib D:\a\perl5\perl5\cpan\AutoLoader\lib D:\a\perl5\perl5\dist\Carp\lib D:\a\perl5\perl5\dist\PathTools D:\a\perl5\perl5\dist\PathTools\lib D:\a\perl5\perl5\cpan\ExtUtils-Install\lib D:\a\perl5\perl5\cpan\ExtUtils-MakeMaker\lib D:\a\perl5\perl5\cpan\ExtUtils-Manifest\lib D:\a\perl5\perl5\cpan\File-Path\lib D:\a\perl5\perl5\ext\re D:\a\perl5\perl5\dist\Term-ReadLine\lib D:\a\perl5\perl5\dist\Exporter\lib D:\a\perl5\perl5\ext\File-Find\lib D:\a\perl5\perl5\cpan\Text-Tabs\lib D:\a\perl5\perl5\dist\constant\lib D:\a\perl5\perl5\cpan\version\lib D:\a\perl5\perl5\cpan\Getopt-Long\lib D:\a\perl5\perl5\cpan\Text-ParseWords\lib D:\a\perl5\perl5\cpan\ExtUtils-PL2Bat\lib D:\a\perl5\perl5\dist\ExtUtils-ParseXS\lib D:\a\perl5\perl5\cpan\parent\lib D:\a\perl5\perl5\cpan\ExtUtils-Constant\lib .) at D:\a\perl5\perl5\cpan\Getopt-Long\lib/Getopt/Long.pm line 23.
2023-11-13T02:04:34.6022680Z BEGIN failed--compilation aborted at D:\a\perl5\perl5\cpan\Getopt-Long\lib/Getopt/Long.pm line 23.
2023-11-13T02:04:34.6027850Z Compilation failed in require at ..\mkppport line 6.
2023-11-13T02:04:34.6030220Z BEGIN failed--compilation aborted at ..\mkppport line 6.
2023-11-13T02:04:34.6043558Z gmake: *** [GNUMakefile:1544: MakePPPort] Error 2
2023-11-13T02:04:34.6044270Z gmake: *** Waiting for unfinished jobs....
2023-11-13T02:04:34.8117353Z "D:\a\perl5\perl5\miniperl.exe" "-I..\..\lib" lib_pm.PL lib.pm 
2023-11-13T02:04:34.8486546Z Extracting lib.pm (with variable substitutions)
2023-11-13T02:04:35.0128788Z gmake[1]: Leaving directory 'D:/a/perl5/perl5/dist/lib'
```
I should note that this version of Getopt-Long is (a) doing fine on CPANtesters; (b) doing fine on our smoke-testing rigs; and (c) doing fine in all other runs on our GH CI rig, **including two Windows MSVC 142 rigs.**  The upstream author (@sciurius) is the long-time maintainer of this distro.

The MingW failures at a simple (new) invocation of `use base qw(Exporter);`.  I don't see why that should cause a Perl build-time failure.  The mixture of forwards- and backwards-path separators in the CI output is suspicious.

I don't know much about our CI rigs and have never personally used MingW.  Could some people who are knowledgeable about either please take a look?

Thank you very much.

--- a/MANIFEST
+++ b/MANIFEST
@@ -1150,12 +1150,15 @@ cpan/Filter-Util-Call/t/call.t			See if Filter::Util::Call works
 cpan/Filter-Util-Call/t/rt_101033.pm
 cpan/Filter-Util-Call/t/rt_101033.t
 cpan/Filter-Util-Call/t/rt_54452-rebless.t
-cpan/Getopt-Long/lib/Getopt/Long.pm	Fetch command options (GetOptions)
-cpan/Getopt-Long/t/gol-basic.t		See if Getopt::Long works
-cpan/Getopt-Long/t/gol-linkage.t	See if Getopt::Long works
-cpan/Getopt-Long/t/gol-oo.t		See if Getopt::Long works
-cpan/Getopt-Long/t/gol-xargv.t		See if Getopt::Long works
-cpan/Getopt-Long/t/gol-xstring.t	See if Getopt::Long works
+cpan/Getopt-Long/lib/Getopt/Long.pm		Fetch command options (GetOptions)
+cpan/Getopt-Long/lib/Getopt/Long/Parser.pm	Getopt-Long
+cpan/Getopt-Long/t/gol-basic.t			See if Getopt::Long works
+cpan/Getopt-Long/t/gol-linkage.t		See if Getopt::Long works
+cpan/Getopt-Long/t/gol-load1.t			Getopt-Long
+cpan/Getopt-Long/t/gol-load2.t			Getopt-Long
+cpan/Getopt-Long/t/gol-oo.t			See if Getopt::Long works
+cpan/Getopt-Long/t/gol-xargv.t			See if Getopt::Long works
+cpan/Getopt-Long/t/gol-xstring.t		See if Getopt::Long works
 cpan/HTTP-Tiny/corpus/auth-01.txt	Test file related to HTTP::Tiny
 cpan/HTTP-Tiny/corpus/auth-02.txt	Test file related to HTTP::Tiny
 cpan/HTTP-Tiny/corpus/auth-03.txt	Test file related to HTTP::Tiny

--- a/Makefile.SH
+++ b/Makefile.SH
@@ -1441,8 +1441,8 @@ _cleaner2:
 	-rmdir lib/IO/Compress/Zlib lib/IO/Compress/Zip lib/IO/Compress/Gzip
 	-rmdir lib/IO/Compress/Base lib/IO/Compress/Adapter lib/IO/Compress
 	-rmdir lib/IO lib/I18N/LangTags lib/I18N lib/Hash/Util lib/Hash
-	-rmdir lib/HTTP lib/Filter/Util lib/Filter lib/File/Spec
-	-rmdir lib/ExtUtils/Typemaps lib/ExtUtils/ParseXS
+	-rmdir lib/HTTP lib/Getopt/Long lib/Filter/Util lib/Filter
+	-rmdir lib/File/Spec lib/ExtUtils/Typemaps lib/ExtUtils/ParseXS
 	-rmdir lib/ExtUtils/MakeMaker/version lib/ExtUtils/MakeMaker
 	-rmdir lib/ExtUtils/Liblist lib/ExtUtils/Constant lib/ExtUtils/Command
 	-rmdir lib/ExtUtils/CBuilder/Platform/Windows

--- a/Porting/Maintainers.pl
+++ b/Porting/Maintainers.pl
@@ -612,7 +612,8 @@ our %Modules = (
     },
 
     'Getopt::Long' => {
-        'DISTRIBUTION' => 'JV/Getopt-Long-2.54.tar.gz',
+        'DISTRIBUTION' => 'JV/Getopt-Long-2.57.tar.gz',
+        'SYNCINFO'     => 'jkeenan on Sat Nov 11 13:09:21 2023',
         'FILES'        => q[cpan/Getopt-Long],
         'EXCLUDED'     => [
             qr{^examples/},

--- a/cpan/Getopt-Long/lib/Getopt/Long.pm
+++ b/cpan/Getopt-Long/lib/Getopt/Long.pm
@@ -3,31 +3,24 @@
 # Getopt::Long.pm -- Universal options parsing
 # Author          : Johan Vromans
 # Created On      : Tue Sep 11 15:00:12 1990
-# Last Modified By: Johan Vromans
-# Last Modified On: Thu Nov 17 17:45:27 2022
-# Update Count    : 1777
+# Last Modified On: Sat Nov 11 17:48:41 2023
+# Update Count    : 1808
 # Status          : Released
 
 ################ Module Preamble ################
 
-# There are no CPAN testers for very old versions of Perl.
-# Getopt::Long is reported to run under 5.8.
-use 5.004;
+# Getopt::Long is reported to run under 5.6.1. Thanks Tux!
+use 5.006001;
 
 use strict;
 use warnings;
 
 package Getopt::Long;
 
-use vars qw($VERSION);
-$VERSION        =  2.54;
-# For testing versions only.
-use vars qw($VERSION_STRING);
-$VERSION_STRING = "2.54";
+our $VERSION = 2.57;
 
 use Exporter;
-use vars qw(@ISA @EXPORT @EXPORT_OK);
-@ISA = qw(Exporter);
+use base qw(Exporter);
 
 # Exported subroutines.
 sub GetOptions(@);		# always
@@ -37,21 +30,24 @@ sub Configure(@);		# on demand
 sub HelpMessage(@);		# on demand
 sub VersionMessage(@);		# in demand
 
+our @EXPORT;
+our @EXPORT_OK;
+# Values for $order. See GNU getopt.c for details.
+our ($REQUIRE_ORDER, $PERMUTE, $RETURN_IN_ORDER);
 BEGIN {
-    # Init immediately so their contents can be used in the 'use vars' below.
+    ($REQUIRE_ORDER, $PERMUTE, $RETURN_IN_ORDER) = (0..2);
     @EXPORT    = qw(&GetOptions $REQUIRE_ORDER $PERMUTE $RETURN_IN_ORDER);
     @EXPORT_OK = qw(&HelpMessage &VersionMessage &Configure
 		    &GetOptionsFromArray &GetOptionsFromString);
 }
 
 # User visible variables.
-use vars @EXPORT, @EXPORT_OK;
-use vars qw($error $debug $major_version $minor_version);
+our ($error, $debug, $major_version, $minor_version);
 # Deprecated visible variables.
-use vars qw($autoabbrev $getopt_compat $ignorecase $bundling $order
-	    $passthrough);
+our ($autoabbrev, $getopt_compat, $ignorecase, $bundling, $order,
+     $passthrough);
 # Official invisible variables.
-use vars qw($genprefix $caller $gnu_compat $auto_help $auto_version $longprefix);
+our ($genprefix, $caller, $gnu_compat, $auto_help, $auto_version, $longprefix);
 
 # Really invisible variables.
 my $bundling_values;
@@ -125,16 +121,10 @@ sub import {
 
 ################ Initialization ################
 
-# Values for $order. See GNU getopt.c for details.
-($REQUIRE_ORDER, $PERMUTE, $RETURN_IN_ORDER) = (0..2);
 # Version major/minor numbers.
 ($major_version, $minor_version) = $VERSION =~ /^(\d+)\.(\d+)/;
 
 ConfigDefaults();
-
-################ OO Interface ################
-
-package Getopt::Long::Parser;
 
 # Store a copy of the default configuration. Since ConfigDefaults has
 # just been called, what we get from Configure is the default.
@@ -142,80 +132,18 @@ my $default_config = do {
     Getopt::Long::Configure ()
 };
 
-sub new {
-    my $that = shift;
-    my $class = ref($that) || $that;
-    my %atts = @_;
-
-    # Register the callers package.
-    my $self = { caller_pkg => (caller)[0] };
-
-    bless ($self, $class);
-
-    # Process config attributes.
-    if ( defined $atts{config} ) {
-	my $save = Getopt::Long::Configure ($default_config, @{$atts{config}});
-	$self->{settings} = Getopt::Long::Configure ($save);
-	delete ($atts{config});
-    }
-    # Else use default config.
-    else {
-	$self->{settings} = $default_config;
-    }
-
-    if ( %atts ) {		# Oops
-	die(__PACKAGE__.": unhandled attributes: ".
-	    join(" ", sort(keys(%atts)))."\n");
-    }
-
-    $self;
-}
-
-sub configure {
-    my ($self) = shift;
-
-    # Restore settings, merge new settings in.
-    my $save = Getopt::Long::Configure ($self->{settings}, @_);
-
-    # Restore orig config and save the new config.
-    $self->{settings} = Getopt::Long::Configure ($save);
-}
-
-sub getoptions {
-    my ($self) = shift;
-
-    return $self->getoptionsfromarray(\@ARGV, @_);
-}
-
-sub getoptionsfromarray {
-    my ($self) = shift;
-
-    # Restore config settings.
-    my $save = Getopt::Long::Configure ($self->{settings});
-
-    # Call main routine.
-    my $ret = 0;
-    $Getopt::Long::caller = $self->{caller_pkg};
-
-    eval {
-	# Locally set exception handler to default, otherwise it will
-	# be called implicitly here, and again explicitly when we try
-	# to deliver the messages.
-	local ($SIG{__DIE__}) = 'DEFAULT';
-	$ret = Getopt::Long::GetOptionsFromArray (@_);
-    };
-
-    # Restore saved settings.
-    Getopt::Long::Configure ($save);
-
-    # Handle errors and return value.
-    die ($@) if $@;
-    return $ret;
-}
-
-package Getopt::Long;
+# For the parser only.
+sub _default_config { $default_config }
 
 ################ Back to Normal ################
+
+# The ooparser was traditionally part of the main package.
+no warnings 'redefine';
+sub Getopt::Long::Parser::new {
+    require Getopt::Long::Parser;
+    goto &Getopt::Long::Parser::new;
+}
+use warnings 'redefine';
 
 # Indices in option control info.
 # Note that ParseOptions uses the fields directly. Search for 'hard-wired'.
@@ -305,7 +233,7 @@ sub GetOptionsFromArray(@) {
 	# Avoid some warnings if debugging.
 	local ($^W) = 0;
 	print STDERR
-	  ("Getopt::Long $Getopt::Long::VERSION_STRING ",
+	  ("Getopt::Long $VERSION ",
 	   "called from package \"$pkg\".",
 	   "\n  ",
 	   "argv: ",
@@ -806,11 +734,15 @@ sub OptCtl ($) {
 sub ParseOptionSpec ($$) {
     my ($opt, $opctl) = @_;
 
+    # Allow period in option name unless passing through,
+    my $op = $passthrough
+      ? qr/(?: \w+[-\w]* )/x : qr/(?: \w+[-.\w]* )/x;
+
     # Match option spec.
     if ( $opt !~ m;^
 		   (
 		     # Option name
-		     (?: \w+[-\w]* )
+		     $op
 		     # Aliases
 		     (?: \| (?: . [^|!+=:]* )? )*
 		   )?
@@ -929,7 +861,8 @@ sub ParseOptionSpec ($$) {
 	}
     }
 
-    if ( $dups && $^W ) {
+    if ( $dups ) {
+	# Warn now. Will become fatal in a future release.
 	foreach ( split(/\n+/, $dups) ) {
 	    warn($_."\n");
 	}
@@ -1495,9 +1428,7 @@ sub VersionMessage(@) {
 	       $0, defined $v ? " version $v" : (),
 	       "\n",
 	       "(", __PACKAGE__, "::", "GetOptions",
-	       " version ",
-	       defined($Getopt::Long::VERSION_STRING)
-	         ? $Getopt::Long::VERSION_STRING : $VERSION, ";",
+	       " version $VERSION,",
 	       " Perl version ",
 	       $] >= 5.006 ? sprintf("%vd", $^V) : $],
 	       ")\n");
@@ -1515,7 +1446,7 @@ sub VersionMessage(@) {
 sub HelpMessage(@) {
     eval {
 	require Pod::Usage;
-	import Pod::Usage;
+	Pod::Usage->import;
 	1;
     } || die("Cannot provide help: cannot load Pod::Usage\n");
 
@@ -1941,7 +1872,9 @@ and the argument specification.
 
 The name specification contains the name of the option, optionally
 followed by a list of alternative names separated by vertical bar
-characters.
+characters. The name is made up of alphanumeric characters, hyphens,
+underscores. If C<pass_through> is disabled, a period is also allowed in
+option names.
 
     length	      option name is "length"
     length|size|l     name is "length", aliases are "size" and "l"
@@ -2048,18 +1981,7 @@ option will be incremented.
 
 =head2 Object oriented interface
 
-Getopt::Long can be used in an object oriented way as well:
-
-    use Getopt::Long;
-    $p = Getopt::Long::Parser->new;
-    $p->configure(...configuration options...);
-    if ($p->getoptions(...options descriptions...)) ...
-    if ($p->getoptionsfromarray( \@array, ...options descriptions...)) ...
-
-Configuration options can be passed to the constructor:
-
-    $p = new Getopt::Long::Parser
-             config => [...configuration options...];
+See L<Getopt::Long::Parser>.
 
 =head2 Callback object
 
@@ -2389,11 +2311,12 @@ POSIXLY_CORRECT has been set, in which case C<getopt_compat> is disabled.
 
 C<gnu_compat> controls whether C<--opt=> is allowed, and what it should
 do. Without C<gnu_compat>, C<--opt=> gives an error. With C<gnu_compat>,
-C<--opt=> will give option C<opt> and empty value.
+C<--opt=> will give option C<opt> an empty value.
 This is the way GNU getopt_long() does it.
 
-Note that C<--opt value> is still accepted, even though GNU
-getopt_long() doesn't.
+Note that for options with optional arguments, C<--opt value> is still
+accepted, even though GNU getopt_long() requires writing C<--opt=value>
+in this case.
 
 =item gnu_getopt
 
@@ -2677,7 +2600,7 @@ When no destination is specified for an option, GetOptions will store
 the resultant value in a global variable named C<opt_>I<XXX>, where
 I<XXX> is the primary name of this option. When a program executes
 under C<use strict> (recommended), these variables must be
-pre-declared with our() or C<use vars>.
+pre-declared with our().
 
     our $opt_length = 0;
     GetOptions ('length=i');	# will store in $opt_length
@@ -2805,7 +2728,7 @@ Johan Vromans <jvromans@squirrel.nl>
 
 =head1 COPYRIGHT AND DISCLAIMER
 
-This program is Copyright 1990,2015 by Johan Vromans.
+This program is Copyright 1990,2015,2023 by Johan Vromans.
 This program is free software; you can redistribute it and/or
 modify it under the terms of the Perl Artistic License or the
 GNU General Public License as published by the Free Software

--- a/cpan/Getopt-Long/lib/Getopt/Long/Parser.pm
+++ b/cpan/Getopt-Long/lib/Getopt/Long/Parser.pm
@@ -1,0 +1,177 @@
+#! perl
+
+# Parser.pm -- Getopt::Long object oriented interface
+# Author          : Johan Vromans
+# Created On      : Thu Nov  9 10:37:00 2023
+# Last Modified On: Sat Nov 11 17:48:49 2023
+# Update Count    : 13
+# Status          : Released
+
+package Getopt::Long::Parser;
+
+our $VERSION = 2.57;
+
+=head1 NAME
+
+Getopt::Long::Parser - Getopt::Long object oriented interface
+
+=head1 SYNOPSIS
+
+    use Getopt::Long::Parser;
+    my $p = Getopt::Long::Parser->new;
+    $p->configure( ...configuration options... );
+    if ( $p->getoptions( ...options descriptions... ) ) ...
+    if ( $p->getoptionsfromarray( \@array, ...options descriptions... ) ) ...
+
+Configuration options can be passed to the constructor:
+
+    my $p = Getopt::Long::Parser->new
+             config => [...configuration options...];
+
+=head1 DESCRIPTION
+
+Getopt::Long::Parser is an object oriented interface to
+L<Getopt::Long>. See its documentation for configuration and use.
+
+Note that Getopt::Long and Getopt::Long::Parser are not object
+oriented. Getopt::Long::Parser emulates an object oriented interface,
+which should be okay for most purposes.
+
+=head1 CONSTRUCTOR
+
+    my $p = Getopt::Long::Parser->new( %options );
+
+The constructor takes an optional hash with parameters.
+
+=over 4
+
+=item config
+
+An array reference with configuration settings.
+See L<Getopt::Long/"Configuring Getopt::Long"> for all possible settings.
+
+=back
+
+=cut
+
+# Getopt::Long has a stub for Getopt::Long::Parser::new.
+use Getopt::Long ();
+no warnings 'redefine';
+
+sub new {
+    my $that = shift;
+    my $class = ref($that) || $that;
+    my %atts = @_;
+
+    # Register the callers package.
+    my $self = { caller_pkg => (caller)[0] };
+
+    bless ($self, $class);
+
+    my $default_config = Getopt::Long::_default_config();
+
+    # Process config attributes.
+    if ( defined $atts{config} ) {
+	my $save = Getopt::Long::Configure ($default_config, @{$atts{config}});
+	$self->{settings} = Getopt::Long::Configure ($save);
+	delete ($atts{config});
+    }
+    # Else use default config.
+    else {
+	$self->{settings} = $default_config;
+    }
+
+    if ( %atts ) {		# Oops
+	die(__PACKAGE__.": unhandled attributes: ".
+	    join(" ", sort(keys(%atts)))."\n");
+    }
+
+    $self;
+}
+
+use warnings 'redefine';
+
+=head1 METHODS
+
+In the examples, $p is assumed to be the result of a call to the constructor.
+
+=head2 configure
+
+    $p->configure( %settings );
+
+Update the current config settings.
+See L<Getopt::Long/"Configuring Getopt::Long"> for all possible settings.
+
+=cut
+
+sub configure {
+    my ($self) = shift;
+
+    # Restore settings, merge new settings in.
+    my $save = Getopt::Long::Configure ($self->{settings}, @_);
+
+    # Restore orig config and save the new config.
+    $self->{settings} = Getopt::Long::Configure ($save);
+}
+
+=head2 getoptionsfromarray
+
+    $res = $p->getoptionsfromarray( $aref, @opts );
+
+=head2 getoptions
+
+    $res = $p->getoptions( @opts );
+
+The same as getoptionsfromarray( \@ARGV, @opts ).
+
+=cut
+
+sub getoptions {
+    my ($self) = shift;
+
+    return $self->getoptionsfromarray(\@ARGV, @_);
+}
+
+sub getoptionsfromarray {
+    my ($self) = shift;
+
+    # Restore config settings.
+    my $save = Getopt::Long::Configure ($self->{settings});
+
+    # Call main routine.
+    my $ret = 0;
+    $Getopt::Long::caller = $self->{caller_pkg};
+
+    eval {
+	# Locally set exception handler to default, otherwise it will
+	# be called implicitly here, and again explicitly when we try
+	# to deliver the messages.
+	local ($SIG{__DIE__}) = 'DEFAULT';
+	$ret = Getopt::Long::GetOptionsFromArray (@_);
+    };
+
+    # Restore saved settings.
+    Getopt::Long::Configure ($save);
+
+    # Handle errors and return value.
+    die ($@) if $@;
+    return $ret;
+}
+
+=head1 SEE ALSO
+
+L<Getopt::Long>
+
+=head1 AUTHOR
+
+Johan Vromans <jvromans@squirrel.nl>
+
+=head1 COPYRIGHT AND DISCLAIMER
+
+This program is Copyright 1990,2015,2023 by Johan Vromans.
+This program is free software; you can redistribute it and/or
+modify it under the same terms as Perl.
+
+=cut
+
+1;

--- a/cpan/Getopt-Long/t/gol-load1.t
+++ b/cpan/Getopt-Long/t/gol-load1.t
@@ -1,0 +1,37 @@
+#! perl
+
+# Verify that loading Getopt::Long does not load Getopt::Long::Parser
+# until/unless used.
+
+BEGIN {
+    if( $ENV{PERL_CORE} ) {
+        chdir 't';
+        @INC = ('../lib', 'lib');
+    }
+    else {
+        unshift @INC, 't/lib';
+    }
+}
+
+use strict;
+use warnings;
+use Test::More tests => 4;
+
+use_ok("Getopt::Long");
+
+# Getopt::Long::Parser should not be loaded.
+ok( !defined $Getopt::Long::Parser::VERSION,
+    "Getopt::Long did not load Parser" );
+
+# Create a parser object.
+my $p = Getopt::Long::Parser->new;
+
+# Getopt::Long::Parser should now be loaded.
+ok( defined $Getopt::Long::Parser::VERSION,
+    "Parser $Getopt::Long::Parser::VERSION loaded" );
+
+# Verify version match.
+is( $Getopt::Long::VERSION, $Getopt::Long::Parser::VERSION,
+    "Parser version matches" );
+
+diag( "Testing Getopt::Long $Getopt::Long::VERSION, Perl $], $^X" );

--- a/cpan/Getopt-Long/t/gol-load2.t
+++ b/cpan/Getopt-Long/t/gol-load2.t
@@ -1,0 +1,27 @@
+#! perl
+
+# Verify that loading Getopt::Long::Parser also loads Getopt::Long.
+
+BEGIN {
+    if( $ENV{PERL_CORE} ) {
+        chdir 't';
+        @INC = ('../lib', 'lib');
+    }
+    else {
+        unshift @INC, 't/lib';
+    }
+}
+
+use strict;
+use warnings;
+use Test::More tests => 3;
+
+use_ok("Getopt::Long::Parser");
+
+# Getopt::Long should have been loaded as well.
+ok( defined $Getopt::Long::VERSION,
+    "Getopt::Long::Parser loads Getopt::Long" );
+
+# Verify version match.
+is( $Getopt::Long::VERSION, $Getopt::Long::Parser::VERSION,
+    "Parser version matches" );

--- a/cpan/Getopt-Long/t/gol-oo.t
+++ b/cpan/Getopt-Long/t/gol-oo.t
@@ -9,11 +9,11 @@ BEGIN {
     }
 }
 
-use Getopt::Long;
+use Getopt::Long::Parser;
 my $want_version="2.24";
 die("Getopt::Long version $want_version required--this is only version ".
-    $Getopt::Long::VERSION)
-  unless $Getopt::Long::VERSION ge $want_version;
+    $Getopt::Long::Parser::VERSION)
+  unless $Getopt::Long::Parser::VERSION ge $want_version;
 print "1..14\n";
 
 @ARGV = qw(-Foo -baR --foo bar);

--- a/lib/.gitignore
+++ b/lib/.gitignore
@@ -98,6 +98,7 @@
 /FindBin.pm
 /GDBM_File.pm
 /Getopt/Long.pm
+/Getopt/Long/
 /HTTP/
 /Hash/
 /I18N/

--- a/win32/GNUmakefile
+++ b/win32/GNUmakefile
@@ -1725,6 +1725,7 @@ distclean: realclean
 	-if exist $(LIBDIR)\ExtUtils\Typemaps rmdir /s /q $(LIBDIR)\ExtUtils\Typemaps
 	-if exist $(LIBDIR)\File\Spec rmdir /s /q $(LIBDIR)\File\Spec
 	-if exist $(LIBDIR)\Filter rmdir /s /q $(LIBDIR)\Filter
+	-if exist $(LIBDIR)\Getopt\Long rmdir /s /q $(LIBDIR)\Getopt\Long
 	-if exist $(LIBDIR)\Hash rmdir /s /q $(LIBDIR)\Hash
 	-if exist $(LIBDIR)\HTTP rmdir /s /q $(LIBDIR)\HTTP
 	-if exist $(LIBDIR)\I18N rmdir /s /q $(LIBDIR)\I18N

--- a/win32/Makefile
+++ b/win32/Makefile
@@ -1242,6 +1242,7 @@ distclean: realclean
 	-if exist $(LIBDIR)\ExtUtils\Typemaps rmdir /s /q $(LIBDIR)\ExtUtils\Typemaps
 	-if exist $(LIBDIR)\File\Spec rmdir /s /q $(LIBDIR)\File\Spec
 	-if exist $(LIBDIR)\Filter rmdir /s /q $(LIBDIR)\Filter
+	-if exist $(LIBDIR)\Getopt\Long rmdir /s /q $(LIBDIR)\Getopt\Long
 	-if exist $(LIBDIR)\Hash rmdir /s /q $(LIBDIR)\Hash
 	-if exist $(LIBDIR)\HTTP rmdir /s /q $(LIBDIR)\HTTP
 	-if exist $(LIBDIR)\I18N rmdir /s /q $(LIBDIR)\I18N

--- a/write_buildcustomize.pl
+++ b/write_buildcustomize.pl
@@ -53,6 +53,7 @@ push @toolchain, qw(
 	dist/ExtUtils-ParseXS/lib
 	cpan/parent/lib
 	cpan/ExtUtils-Constant/lib
+    dist/base/lib
 ) if $^O eq 'MSWin32';
 push @toolchain, 'ext/VMS-Filespec/lib' if $^O eq 'VMS';
 

